distributed/client.py

```diff
@@ -578,14 +597,16 @@ def start(self, asynchronous=None, **kwargs):
                 self._should_close_loop = True
             while not self.loop._running:
                 sleep(0.001)
+
         pc = PeriodicCallback(lambda: None, 1000, io_loop=self.loop)
         self.loop.add_callback(pc.start)
         _set_global_client(self)
-        if asynchronous:
+        self.status = 'connecting'
+
+        if self.asynchronous:
```

Comment:
I must admit I fail to understand the intent of the `.asynchronous` property. If the user passed `asynchronous=True`, we may still start an event loop in a separate thread?